infra/base-images/base-builder/detect_repo.py

```diff
@@ -55,7 +57,14 @@ def main():
   else:
     src_dir = os.environ.get('SRC', '/src')
 
-  for single_dir in os.listdir(src_dir):
+  dirs_to_search = os.listdir(src_dir)
```

Comment:
I'd make a function `get_dirs_to_search` and put 60-66 in it.
I'm sure we'll have to change in the future to support other weird build systems (rust maybe).

Reply:
Fixed.